daemon/event.go

```diff
@@ -92,6 +94,34 @@ func (d *daemon) CreateEvent(req *pb.CreateEventRequest, resp pb.Daemon_CreateEv
 		return fmt.Errorf("user credentials could not found on context %v", err)
 	}
 
+	// checking through eventPool is not good enough since booked events are not added to eventpool
+	isEventExist, err := d.dbClient.IsEventExists(ctx, &pbc.GetEventByTagReq{
+		EventTag: req.Tag,
+		// it will take INVERT condition which means that query from
+		//Running, Suspended and Booked events
+		Status: Closed,
+	})
+	if err != nil {
+		noEventExist := fmt.Errorf("event does not exist or something is wrong: %v", err)
+		resp.Send(&pb.LabStatus{ErrorMessage: noEventExist.Error()})
+		return noEventExist
+	}
+	if isEventExist.IsExist {
+		resp.Send(&pb.LabStatus{ErrorMessage: NotAvailableTag})
+		return fmt.Errorf(NotAvailableTag)
+	}
+	log.Debug().Msgf("Checked existing events through database.")
```

Comment:
```suggestion
	log.Debug().Msgf("Checked existing events through database.")
	log.Debug().Msg("Checked existing events through database.")
```
Maybe add additional fields for the logging like which tag was used when checking

Reply:
Again, these logs are not added in this PR. In my opinion, different issues on the platform should not be touched on a PR which is meant to fix particular issue.

Reply:
Totally fair i just believe that it would be nice to improve the function as a whole when fixing issues caused but it :)